src/axom/core/memory_management.hpp

```diff
@@ -37,13 +41,14 @@ constexpr int DYNAMIC_ALLOCATOR_ID = 0;
  */
 enum class MemorySpace
 {
-  Dynamic,
+  Dynamic,  //!< Refers to Umpire's current default allocator
```

Comment:
What happens in the `Dynamic` case if Axom is not configured with Umpire?

Reply:
We will use malloc and free, like we have always done.  However, getDefaultAllocatorID now returns MALLOC_ALLOCATOR_ID instead of 0.  (We decided that 0 is a bad value because we can't guarantee that it won't clash with an Umpire id.)

Reply:
Can we add what you wrote here to the comment?  The choice of `Dynamic` as a "memory space" confuses me. At least with a more informative comment, it's a bit more clear.